src/server/tiering/external_alloc.cc

```diff
@@ -131,22 +131,30 @@ template <size_t N> size_t FindFirst(const std::bitset<N>& bs) {
 */
 namespace detail {
 
+// Page can be exactly in either these 3 states:
+// 1. unitialized - with no blocks being allocated - segment_inuse will be 0 in that case, 1
+// otherwise.
+// 2. Partly utilized by 1 or more blocks, with available > 0 in that case. It must be present in
+// free_pages_ list then.
+// 3. Fully utilized, with available==0, in that case it's not part of free_pages_ list.
 struct Page {
   std::bitset<256> free_blocks;  // bitmask of free blocks (32 bytes).
   uint8_t id;                    // index inside the Segment.pages array.
 
   // need some mapping function to map from block_size to real_block_size given Page class.
-  BinIdx block_size_bin;
-  uint8_t segment_inuse : 1;  // true if segment allocated this page.
+  BinIdx bin_idx;
+  bool segment_inuse;  // true if segment allocated this page.
+  bool freepages_inuse;
```

Comment:
so why do you need `freepages_inuse` then? 
seems that it's equivalent to `available > 0` ?